### PR TITLE
Provide builder for NetworkAttachmentConfig (closes #520)

### LINKED ISSUE #520: Network attachment for docker service create
This is more a question rather a bug report. I 've just updated my dependencies to use the newly introduced docker swarm support. I 'm struggling to find out how I specify a network name for service create, I think it's impossible with NetworkAttachmentConfig. Moreover, I think NetworkAttachment for TaskSpec is not needed.

Can you please give me some insight?

### PR DESCRIPTION
**Pull request by dmandalidis**

### CODE REVIEW

**Comment by dmandalidis** [comment]
@davidxia I know it's the snapshot version that this PR targets, but I 'd be very grateful if you could give some priority to this and comment or merge. Many thanks in advance

File: src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java [hunk 1]
```diff
@@ -3364,6 +3365,45 @@ public void testInspectSwarm() throws Exception {
...
+              .withName(serviceName)
+              .withTaskTemplate(taskSpec)
+              .withServiceMode(ServiceMode.withReplicas(1L))
+              .withNetworks(NetworkAttachmentConfig.builder().withTarget(networkName).build())
+              .build();
+      
+      final ServiceCreateResponse response = sut.createService(spec, new ServiceCreateOptions());
+      assertThat(response.id(), is(notNullValue()));
+      
+      sut.removeService(serviceName);
```
Comment (davidxia) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: Perhaps we can inspect the service for the correct network?

```java
final Service service = sut.inspectService(response.id());
assertThat(service.spec().networks().size(), equalTo(1));
assertThat(service.spec().networks().get(0).target(), equalTo(networkCreation.id()));

```

File: src/main/java/com/spotify/docker/client/messages/swarm/NetworkAttachmentConfig.java [hunk 2]
```diff
@@ -16,16 +16,16 @@
  */
 package com.spotify.docker.client.messages.swarm;
 
-import com.fasterxml.jackson.annotation.JsonAutoDetect;
-import com.fasterxml.jackson.annotation.JsonProperty;
-import com.google.common.base.MoreObjects;
-import com.google.common.collect.ImmutableList;
+import static com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility.ANY;
```
Comment (davidxia) [inline src/main/java/com/spotify/docker/client/messages/swarm/NetworkAttachmentConfig.java]: Spurious change in order of imports.

Comment (dmandalidis) [inline src/main/java/com/spotify/docker/client/messages/swarm/NetworkAttachmentConfig.java]: Many thanks for the fast response. PR updated.

**Comment by davidxia** [comment]
@dmandalidis Is your commit message accurate?

**Comment by dmandalidis** [comment]
@davidxia yeap, the PR description was inaccurate (updated)

**Pull request merged**